From ee02fe03cc1c9e7a14a002434793386c2d268792 Mon Sep 17 00:00:00 2001
From: phanirithvij <phanirithvij2000@gmail.com>
Date: Sun, 4 Jan 2026 16:07:00 +0530
Subject: [PATCH 1/2] fix git hash

Signed-off-by: phanirithvij <phanirithvij2000@gmail.com>
---
 lib/utils/git-hash.ts | 12 ++----------
 1 file changed, 2 insertions(+), 10 deletions(-)

diff --git a/lib/utils/git-hash.ts b/lib/utils/git-hash.ts
index 458651ac2..795bb8811 100644
--- a/lib/utils/git-hash.ts
+++ b/lib/utils/git-hash.ts
@@ -1,14 +1,6 @@
 import { execSync } from 'node:child_process';
 
-let gitHash = process.env.HEROKU_SLUG_COMMIT?.slice(0, 8) || process.env.VERCEL_GIT_COMMIT_SHA?.slice(0, 8);
-let gitDate: Date | undefined;
-if (!gitHash) {
-    try {
-        gitHash = execSync('git rev-parse HEAD').toString().trim().slice(0, 8);
-        gitDate = new Date(execSync('git log -1 --format=%cd').toString().trim());
-    } catch {
-        gitHash = 'unknown';
-    }
-}
+let gitHash = '@GIT_HASH@'.slice(0, 8);
+let gitDate = new Date('Thu Jan 1 00:00:00 1970 +0000');
 
 export { gitDate, gitHash };
-- 
2.51.2

