From bdf183bbd00ae7c2d33bafb17da68ceac71cbadd Mon Sep 17 00:00:00 2001
From: phanirithvij <phanirithvij2000@gmail.com>
Date: Sun, 4 Jan 2026 16:06:13 +0530
Subject: [PATCH 2/2] fix network call

Signed-off-by: phanirithvij <phanirithvij2000@gmail.com>
---
 scripts/workflow/build-routes.ts | 3 +++
 1 file changed, 3 insertions(+)

diff --git a/scripts/workflow/build-routes.ts b/scripts/workflow/build-routes.ts
index a9cb72b21..9446fe077 100644
--- a/scripts/workflow/build-routes.ts
+++ b/scripts/workflow/build-routes.ts
@@ -1,5 +1,6 @@
 import fs from 'node:fs';
 import path from 'node:path';
+import { exit } from 'node:process';
 
 import { parse } from 'tldts';
 import toSource from 'tosource';
@@ -113,3 +114,5 @@ if (isWorkerBuild) {
     fs.writeFileSync(path.join(__dirname, '../../assets/build/routes.js'), `export default ${JSON.stringify(namespaces, null, 2)}`.replaceAll(/"module": "(.*)"\n/g, `"module": $1\n`));
     fs.writeFileSync(path.join(__dirname, '../../assets/build/route-paths.ts'), routePathsType);
 }
+
+exit(0);
-- 
2.51.2

