commit a4afe26ba2cab14535522519ce48b31e00cb1438
Author: phanirithvij <phanirithvij2000@gmail.com>
Date:   Tue Aug 18 10:42:45 2026 +0530

    Allow filtering worktrees by branch name in Worktrees pane

diff --git a/pkg/gui/context/worktrees_context.go b/pkg/gui/context/worktrees_context.go
index 3e45f2d45..690fa6d4b 100644
--- a/pkg/gui/context/worktrees_context.go
+++ b/pkg/gui/context/worktrees_context.go
@@ -16,8 +16,8 @@ var _ types.IListContext = (*WorktreesContext)(nil)
 func NewWorktreesContext(c *ContextCommon) *WorktreesContext {
 	viewModel := NewFilteredListViewModel(
 		func() []*models.Worktree { return c.Model().Worktrees },
-		func(Worktree *models.Worktree) []string {
-			return []string{Worktree.Name}
+		func(worktree *models.Worktree) []string {
+			return []string{worktree.Name, worktree.Branch}
 		},
 	)
 
