diff --git a/ui/app/views/viewactions.go b/ui/app/views/viewactions.go
index 2812791..455c075 100644
--- a/ui/app/views/viewactions.go
+++ b/ui/app/views/viewactions.go
@@ -116,8 +116,10 @@ func (v *viewActions) power(set ...string) bool {
 	}

 	if err := v.rv.adapter.currentSession().SetPoweredState(!powered); err != nil {
-		v.rv.status.ErrorMessage(errors.New("cannot set adapter power state"))
-		return false
+		if err = setBluetoothPower(!powered); err != nil {
+			v.rv.status.ErrorMessage(errors.New("cannot set adapter power state"))
+			return false
+		}
 	}

 	if powered {
diff --git a/ui/app/views/viewactionshelper.go b/ui/app/views/viewactionshelper.go
new file mode 100644
index 0000000..045ccc0
--- /dev/null
+++ b/ui/app/views/viewactionshelper.go
@@ -0,0 +1,20 @@
+package views
+
+import "os/exec"
+
+func setBluetoothPower(enable bool) (err error) {
+	_, err = exec.LookPath("rfkill")
+	if err != nil {
+		return
+	}
+	action := "block"
+	if enable {
+		action = "unblock"
+	}
+	cmd := exec.Command("sh", "-c", "rfkill "+action+" bluetooth")
+	err = cmd.Run()
+	if err != nil {
+		return
+	}
+	return
+}
