From 11dc08d0dc9c21963a0de67831aa87a32400ae0e Mon Sep 17 00:00:00 2001
From: phanirithvij <phanirithvij2000@gmail.com>
Date: Sun, 23 Aug 2026 22:32:57 +0530
Subject: [PATCH] fix: prevent clap from panicking on test harness arguments

---
 src/config/cli.rs | 11 ++++++++++-
 1 file changed, 10 insertions(+), 1 deletion(-)

diff --git a/src/config/cli.rs b/src/config/cli.rs
index c6f86f9..a2adc25 100644
--- a/src/config/cli.rs
+++ b/src/config/cli.rs
@@ -90,7 +90,16 @@ pub(super) struct ClapConfig {
 
 impl ClapConfig {
     pub fn new() -> Self {
-        Self::parse()
+        match Self::try_parse() {
+            Ok(config) => config,
+            Err(e) => {
+                if cfg!(test) {
+                    Self::parse_from(vec!["navi"])
+                } else {
+                    e.exit()
+                }
+            }
+        }
     }
 }
 
-- 
2.55.0

